ansibullbot/triagers/plugins/needs_revision.py

```diff
@@ -607,3 +607,66 @@ def get_shippable_run_facts(iw, meta, shippable=None):
     }
 
     return rmeta
+
+
+def get_last_shippable_full_run_date(ci_status, shippable):
+    '''Map partial re-runs back to their last full run date'''
+
+    # https://github.com/ansible/ansibullbot/issues/935
+
+    # (Epdb) pp [x['target_url'] for x in ci_status]
+    # [u'https://app.shippable.com/github/ansible/ansible/runs/67039/summary',
+    # u'https://app.shippable.com/github/ansible/ansible/runs/67039/summary',
+    # u'https://app.shippable.com/github/ansible/ansible/runs/67039',
+    # u'https://app.shippable.com/github/ansible/ansible/runs/67037/summary',
+    # u'https://app.shippable.com/github/ansible/ansible/runs/67037/summary',
+    # u'https://app.shippable.com/github/ansible/ansible/runs/67037']
+
+    if shippable is None:
+        return None
+
+    # extract and unique the jobids from the target urls
+    runids = ci_status[:]
+    runids = [x['target_url'] for x in runids if 'shippable.com' in x['target_url']]
+    for idx, x in enumerate(runids):
+        paths = x.split('/')
+        if paths[-1].isdigit():
+            runids[idx] = int(paths[-1])
+        elif paths[-2].isdigit():
+            runids[idx] = int(paths[-2])
+
+    # get rid of duplicates and sort
+    runids = sorted(set(runids))
+
+    # always use the numerically higher run id
+    runid = runids[-1]
+
+    # build a datastructure to hold the info collected
+    rundata = {
+        'runid': runid,
+        'created_at': None,
+        'rerun_batch_id': None,
+        'rerun_batch_createdat': None
+    }
+
+    # query the api for all data on this runid
+    rdata = shippable.get_run_data(str(runid), usecache=False)
+
+    # get the referenced run for the last runid if it exists
+    rundata['rerun_batch_id'] = rdata.get('reRunBatchId')
+
+    # keep the timestamp too
+    rundata['created_at'] = rdata.get('createdAt')
+
+    # if it had a rerunbatchid it was a partial run and
+    # we need to go get the date on the original run
+    if rundata['rerun_batch_id']:
```

Comment:
This needs to loop until there are no more re-runs. Otherwise a re-run of a re-run will pick up the wrong date.